builtin/providers/aws/resource_aws_codepipeline.go

```diff
@@ -0,0 +1,504 @@
+			if ok && actionProvider == "GitHub" {
+				delete(config, "OAuthToken")
+			}
+			values["configuration"] = config
+		}
+
+		if len(action.OutputArtifacts) > 0 {
+			values["output_artifacts"] = flattenAwsCodePipelineActionsOutputArtifacts(action.OutputArtifacts)
+		}
+
+		if len(action.InputArtifacts) > 0 {
+			values["input_artifacts"] = flattenAwsCodePipelineActionsInputArtifacts(action.InputArtifacts)
+		}
+
+		if action.RunOrder != nil {
+			values["run_order"] = int(*action.RunOrder)
+		}
+
+		actionsList = append(actionsList, values)
+	}
+	return actionsList
+}
+
+func expandAwsCodePipelineStageActionConfiguration(config map[string]interface{}) map[string]*string {
+	m := map[string]*string{}
+	for k, v := range config {
+		s := v.(string)
+		m[k] = &s
+	}
+	return m
+}
+
+func flattenAwsCodePipelineStageActionConfiguration(config map[string]*string) map[string]string {
+	m := map[string]string{}
+	for k, v := range config {
+		m[k] = *v
+	}
+	return m
+}
+
+func expandAwsCodePipelineActionsOutputArtifacts(s []interface{}) []*codepipeline.OutputArtifact {
+	outputArtifacts := []*codepipeline.OutputArtifact{}
+	for _, artifact := range s {
+		outputArtifacts = append(outputArtifacts, &codepipeline.OutputArtifact{
+			Name: aws.String(artifact.(string)),
+		})
+	}
+	return outputArtifacts
+}
+
+func flattenAwsCodePipelineActionsOutputArtifacts(artifacts []*codepipeline.OutputArtifact) []string {
+	values := []string{}
+	for _, artifact := range artifacts {
+		values = append(values, *artifact.Name)
+	}
+	return values
+}
+
+func expandAwsCodePipelineActionsInputArtifacts(s []interface{}) []*codepipeline.InputArtifact {
+	outputArtifacts := []*codepipeline.InputArtifact{}
+	for _, artifact := range s {
+		outputArtifacts = append(outputArtifacts, &codepipeline.InputArtifact{
+			Name: aws.String(artifact.(string)),
+		})
+	}
+	return outputArtifacts
+}
+
+func flattenAwsCodePipelineActionsInputArtifacts(artifacts []*codepipeline.InputArtifact) []string {
+	values := []string{}
+	for _, artifact := range artifacts {
+		values = append(values, *artifact.Name)
+	}
+	return values
+}
+
+func resourceAwsCodePipelineRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).codepipelineconn
+	resp, err := conn.GetPipeline(&codepipeline.GetPipelineInput{
+		Name: aws.String(d.Id()),
+	})
+
+	if err != nil {
+		return fmt.Errorf("[ERROR] Error retreiving Pipeline: %q", err)
+	}
+	pipeline := resp.Pipeline
+
+	if err := d.Set("artifact_store", flattenAwsCodePipelineArtifactStore(pipeline.ArtifactStore)); err != nil {
+		return err
+	}
+
+	if err := d.Set("stage", flattenAwsCodePipelineStages(pipeline.Stages)); err != nil {
+		return err
+	}
+
+	d.Set("name", pipeline.Name)
+	d.Set("role_arn", pipeline.RoleArn)
+	return nil
+}
+
+func resourceAwsCodePipelineUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).codepipelineconn
+
+	pipeline := expandAwsCodePipeline(d)
```

Comment:
It appears I can't. Update takes a PipelineDeclaration which is the same as the create, it has a lot of required fields.

http://docs.aws.amazon.com/sdk-for-go/api/service/codepipeline/#UpdatePipelineInput